macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,262 @@
+        32.70, 34.65, 36.71, 38.89, 41.20, 43.65, 46.25, 49.00,
+        51.91, 55.00, 58.27, 61.74, 65.41, 69.30, 73.42, 77.78,
+        82.41, 87.31, 92.50, 98.00, 103.83, 110.00, 116.54, 123.47,
+        130.81, 138.59, 146.83, 155.56, 164.81, 174.61, 185.00, 196.00,
+        207.65, 220.00, 233.08, 246.94, 261.63, 277.18, 293.66, 311.13,
+        329.63, 349.23, 369.99, 392.00, 415.30, 440.00, 466.16, 493.88,
+        523.25, 554.37, 587.33, 622.25, 659.26, 698.46, 739.99, 783.99,
+        830.61, 880.00, 932.33, 987.77, 1046.50, 1108.73, 1174.66, 1244.51,
+        1318.51, 1396.91, 1479.98, 1567.98, 1661.22, 1760.00, 1864.66, 1975.53,
+        2093.00, 2217.46, 2349.32, 2489.02, 2637.02, 2793.83, 2959.96, 3135.96,
+        3322.44, 3520.00, 3729.31, 3951.07, 4186.01, 4434.92, 4698.64, 4978.03,
+        5274.04, 5587.65, 5919.91, 6271.93, 6644.88, 7040.00, 7458.62, 7902.13,
+        8372.02, 8869.84, 9397.27, 9956.06, 10548.08, 11175.30, 11839.82, 12543.85
+      ]
+    doc: |
+      Lookup table to convert a MIDI note into a frequency in hz
+      The lookup table represents the formula (2 ** ((midi_note - 69) / 12)) * 440
+    doc-ref: https://en.wikipedia.org/wiki/MIDI_tuning_standard
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header:
+        pos: param2
+        type: sound_header
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+  sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_samples
+        -orig-id: length
+        type: u4
+        doc: number of samples
+        if: sound_header_type == sound_header_type::standard
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+        if: sound_header_type != sound_header_type::standard
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: midi_note
+        type: u1
+        doc: base frequency of sample, expressed as MIDI note values, 60 is middle C
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+        if: sound_header_type != sound_header_type::standard
```

Comment:
I'd prefer positive matches than negative:

```suggestion
        if: |
          sound_header_type == sound_header_type::extended
          or sound_header_type == sound_header_type::compressed
```

Repeating this for multiple fields is inconvenient, so it might be a good idea to extract it to a value instance `is_extended_or_compressed` (or whatever), or move the fields to a new "extension"

The reason behind this is that it just _explicitly_ says what we know and nothing more: we shouldn't assume anything about the headers we are not aware of. If Apple released a new `sound_header` version and someone added support for it to this KSY spec, it would be _implicitly_ matched by the `if: sound_header_type != sound_header_type::standard` condition, which might be wrong.